text/2018-12-19-unified-log-format.md

```diff
@@ -0,0 +1,239 @@
+# Unified Log Format
+
+## Summary
+
+This RFC stipulates a unified log format called "TiDB Log Format" to be used
+across all TiDB software, i.e. TiDB, TiKV, and PD.
+
+## Motivation
+
+A unified log format makes it easy to collect (i.e. Fluentd) logs into a central
+storage (i.e. ElasticSearch) and allows users to query logs in a structural way
+later.
+
+## Detailed Design
+
+A TiDB Log Format file contains a sequence of *lines* containing UTF-8
+characters terminated by either the sequence LF or CRLF. Each line contains a
+*Log Header Section*, a *Log Message Section* and a *Log Fields Section*,
+concatenated by one whitespace character U+0020 (SPACE).
+
+### Log Header Section
+
+The Log Header Section is in the following format:
+
+```text
+[date_time] [LEVEL] [source_file:line_number]
+```
+
+- `date_time`: The human readable time that the log line is generated in the
+  local time zone in the following format using the [UTS #35] symbol table:
+
+  ```text
+  yyyy/MM/dd HH:mm:ss.SSS ZZZZZ
+  ```
+
+   Sample: `2018/12/15 14:20:11.015 +08:00`
+
+- `LEVEL`: The log level in upper case. Available levels are `FATAL`, `ERROR`,
```

Comment:
Do we need to add the definition / meaning of each level here?

Reply:
Nop, this RFC is only for formats. We have another internal doc stipulates what kind of logs are in what level.